core/src/main/java/org/commonjava/indy/core/ctl/AdminController.java

```diff
@@ -179,4 +180,20 @@ public ArtifactStoreValidateData validateStore(ArtifactStore artifactStore) thro
         return storeValidator.validate(artifactStore);
     }
 
+    public List<ArtifactStore> getDisabledRemoteRepositories() {
+        ArrayList<ArtifactStore> disabledArtifactStores = new ArrayList<>();
+        try {
+            List<ArtifactStore> allRepositories = storeManager.query().getAll();
```

Comment:
I think we need to have a query in the ArtifactStoreQuery API that will allow us to select for enabled == false and type == remote (two fluent-API calls) with something like a getAll() at the end.

The call you have currently will give us little choice but to do a table scan.

Reply:
In ArtifactStoreQuery we have getAll( Predicate<ArtifactStore> filter ) method who is allowing us to filter out based on internal state of ArtifactStore object - can we use that>

Reply:
Yeah, I guess my other concerns will wait for a future refactor of that class...it's a good point that the Predicate-driven filtering is already there (which IMO means that we need a deeper refactor, since Predicate is something that executes on results we've already retrieved, which means table scanning).